### ISSUE TITLE
Single finger scroll

### ISSUE THREAD

**Issue by digikar99** [iiordanov/remote-desktop-clients#138]
Hi! Thanks for the remarkable app.

Is it possible to include an option for single finger scroll - the way other Android apps behave? It seems much more intuitive.

**Comment by iiordanov** [comment]
There is a single handed input mode you can switch to to try. Find it under
input modes in the menu when connected.

What is the behavior of single finger scroll on other apps?

Iordan

**Comment by Shubhamkar** [comment]
Single finger scroll as in, say, WPS Office, or Chrome or Firefox. 

If I have understood the use of the app correctly, then, the single handed mode requires a long click, and then the scroll is enabled, which is unintuitive, in comparison to the other Android apps.

**Comment by iiordanov** [comment]
Well, this app does quite a bit more than a browser with touch gestures,
and therefore one would have to be more creative with what initiates the
various single-finger gestures.

Do you have suggestions on what would initiate scrolling versus zooming
in/out?

**Comment by Shubhamkar** [comment]
Yes, I do get that many more interactions are involved. 

I can imagine the following to work for 'Fit to Screen' mode:
- single finger swipe works as scroll, if a scroll is possible; if a scroll is not possible, and if the finger lies on the title bars of windows, initiate a window move; else nothing happens
- long touch initiates a drag; as in text selection, etc; single finger swipe following long touch acts as left-mouse-click down + mouse move; basically what is now assigned to single finger swipe (except for window move) is assigned to this
- right click remains the same as is presently

Basically, currently, left click seems to be initiated when the finger is down. Instead, let it be initiated when the finger is down and up within a small duration (100ms?). Otherwise a scroll event is initiated in the window with the finger.

I'd like to work on this. (Just completed my first year of a 4 year CS program.) [This](https://github.com/iiordanov/remote-desktop-clients/blob/8c110531d7ab1dc746de80a30427a8eb7c90cfbe/Opaque/src/main/java/com/undatech/opaque/input/InputHandlerDirectDragPan.java) seems to be relevant file. Though I also need to learn Java. Is that the relevant file?

**Comment by digikar99** [comment]
I think I have the time to work on it. So, do let me know if you / somebody is working on this, to avoid redundant work.

**Comment by iiordanov** [comment]
Hello,

Nobody is working on this particular request at the moment.

Thanks!
Iordan

**Comment by digikar99** [comment]
It was nice to give this a try: (I see your point now.)

- LongTap functionality: it's trivial to replace the initiation of the Pan mode with a right click. Personally, I don't require the Pan mode; however, the needs of others may be different.
- I realise that the methods I suggested are far too complicated, and might need work on the server side as well!
- For single finger scroll: it is possible to implement scroll using single finger swipe. However, this runs into problems with the dragMode. A workaround is to implement a manual switch, which determines whether dragMode is turned on. Whether to implement this globally, or only within tablet-mode remains a question.
- For scroll vs drag: I had plans for implementing longTap followed immediately by swipe as drag; while just a swipe as scroll. However, I don't see any apps implementing such methods, suggesting the implementation for this is likely not worth the effort.

To sum up, should such a switch (mentioned in point 2) be implemented globally?

**Comment by iiordanov** [comment]
No to the switch, if it will affect every other input mode or be visible if
people are not using the new input mode. If you can think of an elegant way
for the switch to be visible only when your input mode is selected, I would
have to see what it would look like and how it would behave to decide.

**Comment by digikar99** [comment]
#151 Tablet Mode added.

No such switch added. The UI is good enough for me. I'd be willing to listen to other people about whether such a switch is necessary.

It was fun to work on it. Thanks for the app!
